linera-views/src/backends/rocks_db.rs

```diff
@@ -193,55 +192,35 @@ impl RocksDbStoreExecutor {
-        // Thus in order to have the system working, we need to handle the unlikely case of having to
-        // delete a key starting with [255, ...., 255]
-        let len = batch.operations.len();
-        let mut keys = Vec::new();
-        for i in 0..len {
-            let op = batch.operations.get(i).unwrap();
-            if let WriteOperation::DeletePrefix { key_prefix } = op {
-                if get_upper_bound(key_prefix) == Bound::Unbounded {
-                    for short_key in self.find_keys_by_prefix_internal(key_prefix.to_vec())? {
-                        let mut full_key = self.root_key.clone();
-                        full_key.extend(key_prefix);
-                        full_key.extend(short_key);
-                        keys.push(full_key);
-                    }
-                }
-            }
-        }
-        for key in keys {
-            batch.operations.push(WriteOperation::Delete { key });
-        }
+    fn write_batch_internal(&self, batch: Batch) -> Result<(), RocksDbStoreInternalError> {
         let mut inner_batch = rocksdb::WriteBatchWithTransaction::default();
         for operation in batch.operations {
             match operation {
                 WriteOperation::Delete { key } => {
                     check_key_size(&key)?;
-                    let mut full_key = self.root_key.to_vec();
+                    let mut full_key = self.start_key.to_vec();
                     full_key.extend(key);
                     inner_batch.delete(&full_key)
                 }
                 WriteOperation::Put { key, value } => {
                     check_key_size(&key)?;
-                    let mut full_key = self.root_key.to_vec();
+                    let mut full_key = self.start_key.to_vec();
                     full_key.extend(key);
                     inner_batch.put(&full_key, value)
                 }
                 WriteOperation::DeletePrefix { key_prefix } => {
                     check_key_size(&key_prefix)?;
-                    if let Excluded(upper_bound) = get_upper_bound(&key_prefix) {
-                        let mut full_key1 = self.root_key.to_vec();
-                        full_key1.extend(key_prefix);
-                        let mut full_key2 = self.root_key.to_vec();
-                        full_key2.extend(upper_bound);
-                        inner_batch.delete_range(&full_key1, &full_key2);
-                    }
+                    let mut full_key1 = self.start_key.to_vec();
+                    full_key1.extend(&key_prefix);
+                    let full_key2 =
+                        get_upper_bound_option(&full_key1).expect("the first entry cannot be 255");
+                    inner_batch.delete_range(&full_key1, &full_key2);
                 }
             }
         }
+        let mut full_key = self.start_key.to_vec();
+        full_key[0] = STORED_ROOT_KEYS_PREFIX;
+        inner_batch.put(&full_key, vec![]);
```

Comment:
Same. Do we have to write this every time?

Reply:
That one is called "keys_prefix" lol

Reply:
Corrected. Now we write every time only for IndexedDB.